sympy/integrals/deltafunctions.py

```diff
@@ -172,5 +168,9 @@ def deltaintegrate(f, x):
                     dg, rest_mult_2 = change_mul(dg, x)
                     rest_mult = rest_mult*rest_mult_2
                 point = solve(dg.args[0], x)[0]
-                return (rest_mult.subs(x, point)*Heaviside(x - point))
+                try:
+                    n = dg.args[1]
+                except:
```

Comment:
Please don't use base except. What situation is this code trying to account for?